subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/PmdPlugin.java

```diff
@@ -146,7 +155,15 @@ protected void configureForSourceSet(final SourceSet sourceSet, final Pmd task)
         task.setDescription("Run PMD analysis for " + sourceSet.getName() + " classes");
         task.setSource(sourceSet.getAllJava());
         ConventionMapping taskMapping = task.getConventionMapping();
-        taskMapping.map("classpath", () ->
-            sourceSet.getOutput().plus(sourceSet.getCompileClasspath()));
+        ConfigurationContainer configurations = project.getConfigurations();
+        Configuration compileConfiguration = configurations.getByName(sourceSet.getCompileClasspathConfigurationName());
+        Configuration runtimeConfiguration = configurations.getByName(sourceSet.getRuntimeClasspathConfigurationName());
+        Configuration pmdAuxClasspath = configurations.create(sourceSet.getName() + "PmdAuxClasspath");
+        pmdAuxClasspath.extendsFrom(compileConfiguration, runtimeConfiguration);
+        pmdAuxClasspath.setCanBeConsumed(false);
+        pmdAuxClasspath.setVisible(false);
+        pmdAuxClasspath.shouldResolveConsistentlyWith(compileConfiguration);
```

Comment:
I would only do that if the option has been activated in the `java` extension

Reply:
We've discussed it and decided that we won't go in this direction now. I'll file an issue to track necessary improvements in the consistent resolution API.